### PR TITLE
feat: add pre-commit and update readme

### LINKED ISSUE #77: Set up pre-commit
I think I ran black on the file, should probably set up pre-commit for this project ...

_Originally posted by @bbelderbos in https://github.com/PyBites-Open-Source/pybitesbooks/pull/75#discussion_r1222724876_

### PR DESCRIPTION
**Pull request by niltonpimentel02**
Hey, @bbelderbos this is related to issue #77 I didn't ask to work on it but I worked anyway 🤣 tell me if you find it interesting to merge or if I need to change something.

close #77

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[tool.ruff]
```
Comment (bbelderbos) [inline pyproject.toml:1]: Can you add the other fields to pyproject.toml as well please? You can use uv init (might pull in this issue #99 )

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+extend-exclude = [
+    "*/migrations/*",
+    "*/templates/*",
+    "*/static/*",
+    "*.html",
+    "*.css",
+    "*.js",
+    "__init__.py",
+]
+force-exclude = true
```
Comment (bbelderbos) [inline pyproject.toml:12]: What does this do?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  # Ruff version.
+  rev: v0.6.9
+  hooks:
+    # Run the linter.
+    - id: ruff
+      types_or: [ python, pyi ]
+      args: [ --fix ]
+    # Run the formatter.
+    - id: ruff-format
+      types_or: [ python, pyi ]
```
Comment (bbelderbos) [inline .pre-commit-config.yaml:19]: Why pyi here?

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+[tool.ruff]
+# In addition to the standard set of exclusions, omit all tests, plus a specific file.
```
Comment (bbelderbos) [inline pyproject.toml:2]: I think test modules are still included and they should right?

**Review by bbelderbos — COMMENTED** [review]
Thanks, this is useful. Few small nitpicks, can you check please?

File: pyproject.toml (hunk 2 above)
Comment (niltonpimentel02) [reply pyproject.toml:12]: im gonna quote the docs to be clear:

Whether to enforce exclude and extend-exclude patterns, even for paths that are passed to Ruff explicitly. Typically, Ruff will lint any paths passed in directly, even if they would typically be excluded. Setting force-exclude = true will cause Ruff to respect these exclusions unequivocally.

This is useful for pre-commit, which explicitly passes all changed files to the ruff-pre-commit plugin, regardless of whether they're marked as excluded by Ruff's own settings.

File: pyproject.toml (hunk 1 above)
Comment (niltonpimentel02) [reply pyproject.toml:1]: yes i can work on that issue after this one

File: .pre-commit-config.yaml (hunk 3 above)
Comment (niltonpimentel02) [reply .pre-commit-config.yaml:19]: default configs for ruff-format

File: pyproject.toml (hunk 4 above)
Comment (niltonpimentel02) [reply pyproject.toml:2]: yes, correct!

File: pyproject.toml (hunk 1 above)
Comment (bbelderbos) [reply pyproject.toml:1]: That would be awesome, thanks.

File: pyproject.toml (hunk 2 above)
Comment (bbelderbos) [reply pyproject.toml:12]: ok cool, thanks

**Review by bbelderbos — APPROVED** [review]
Awesome, thanks

**Comment by bbelderbos** [comment]
@niltonpimentel02 the hacktoberfest tag on the repo is enough for this to count for you, right?

**Comment by bbelderbos** [comment]
Do we need to fix the pre-commit version? https://github.com/PyBites-Open-Source/pybitesbooks/actions/runs/11570097551/job/32205164381

**Comment by niltonpimentel02** [comment]
> Do we need to fix the pre-commit version? https://github.com/PyBites-Open-Source/pybitesbooks/actions/runs/11570097551/job/32205164381

we need at least Python 3.9 (that's the version that I'm running locally) on your GitHub actions the version is 3.8 so we need to upgrade it 😄

**Comment by bbelderbos** [comment]
OK thanks, I can do that but as you're already going to work on uv, maybe you want to do that there? 🙏

**Comment by niltonpimentel02** [comment]
> OK thanks, I can do that but as you're already going to work on uv, maybe you want to do that there? 🙏

sure... no problem!

**Pull request merged**